lib/src/sql/statements/show.rs

```diff
@@ -0,0 +1,188 @@
+	}
+}
+
+#[allow(unused)]
+pub fn table_or_database(i: &str) -> IResult<&str, Option<String>> {
+	let (i, v) = alt((
+		map(preceded(tag_no_case("table"), preceded(shouldbespace, table)), |v: Table| Some(v.0)),
+		map(tag_no_case("database"), |_| None),
+	))(i)?;
+	Ok((i, v))
+}
+
+#[allow(unused)]
+pub fn int_str(i: &str) -> IResult<&str, &str> {
+	use nom::{
+		character::complete::{char, one_of},
+		multi::{many0, many1},
+		sequence::terminated,
+	};
+
+	recognize(many1(terminated(one_of("0123456789"), many0(char('_')))))(i)
+}
+
+#[allow(unused)]
+pub fn since(i: &str) -> IResult<&str, u64> {
+	use std::num::ParseIntError;
+
+	let (i, _) = tag_no_case("SINCE")(i)?;
+	let (i, _) = shouldbespace(i)?;
+
+	take_u64(i)
+}
+
+#[allow(unused)]
+pub fn limit(i: &str) -> IResult<&str, u32> {
+	use std::num::ParseIntError;
+
+	let (i, _) = tag_no_case("LIMIT")(i)?;
+	let (i, _) = shouldbespace(i)?;
+
+	take_u32(i)
+}
+
+#[allow(unused)]
+pub fn show(i: &str) -> IResult<&str, ShowStatement> {
+	let (i, _) = tag_no_case("SHOW CHANGES")(i)?;
+	let (i, _) = shouldbespace(i)?;
+	let (i, _) = tag_no_case("FOR")(i)?;
+	let (i, _) = shouldbespace(i)?;
+	let (i, table) = table_or_database(i)?;
+	let (i, since) = opt(preceded(shouldbespace, since))(i)?;
+	let (i, limit) = opt(preceded(shouldbespace, limit))(i)?;
+	Ok((
+		i,
+		ShowStatement {
+			table,
+			since,
+			limit,
+		},
+	))
+}
+
+#[cfg(test)]
+mod tests {
+	use super::*;
+
+	#[test]
+	fn tb() {
+		let sql = "TABLE person";
+		let res = table_or_database(sql);
+		assert!(res.is_ok());
+		let out = res.unwrap().1.unwrap();
+		assert_eq!("person", format!("{}", out))
+	}
+
+	#[test]
+	fn db() {
+		let sql = "DATABASE";
+		let res = table_or_database(sql);
+		assert!(res.is_ok());
+		assert!(res.unwrap().1.is_none())
+	}
+
+	#[test]
+	fn show_table_changes() {
+		let sql = "SHOW CHANGES FOR TABLE person";
+		let res = show(sql);
+		assert!(res.is_ok());
+		let out = res.unwrap().1;
+		assert_eq!(sql, format!("{}", out))
+	}
+
+	#[test]
+	fn show_table_changes_since() {
+		let sql = "SHOW CHANGES FOR TABLE person SINCE 0";
+		let res = show(sql);
+		assert!(res.is_ok());
+		let out = res.unwrap().1;
+		assert_eq!(sql, format!("{}", out))
+	}
+
+	#[test]
+	fn show_table_changes_since_limit() {
+		let sql = "SHOW CHANGES FOR TABLE person SINCE 0 LIMIT 10";
+		let res = show(sql);
+		assert!(res.is_ok());
+		let out = res.unwrap().1;
+		assert_eq!(sql, format!("{}", out))
+	}
+
+	#[test]
+	fn show_database_changes() {
+		let sql = "SHOW CHANGES FOR DATABASE";
+		let res = show(sql);
+		assert!(res.is_ok());
+		let out = res.unwrap().1;
+		assert_eq!(sql, format!("{}", out))
+	}
+
+	#[test]
+	fn show_database_changes_since() {
+		let sql = "SHOW CHANGES FOR DATABASE SINCE 0";
+		let res = show(sql);
+		assert!(res.is_ok());
+		let out = res.unwrap().1;
+		assert_eq!(sql, format!("{}", out))
+	}
+
+	#[test]
+	fn show_database_changes_since_limit() {
+		let sql = "SHOW CHANGES FOR DATABASE SINCE 0 LIMIT 10";
```

Comment:
Shall we add a test without `since`? : 
```sql
SHOW CHANGES FOR DATABASE LIMIT 10
```

Reply:
@emmanuel-keller Hey! Does this one exists in the same source work? https://github.com/surrealdb/surrealdb/blob/f033f2208859cfffd344fd4ff9c7d23aea969807/lib/src/sql/statements/show.rs#L163

Reply:
Sorry, I can't see it. I mean't something like this:

```rust
#[test]
	fn show_database_changes_limit() {
		let sql = "SHOW CHANGES FOR DATABASE LIMIT 10";
		let res = show(sql);
		assert!(res.is_ok());
		let out = res.unwrap().1;
		assert_eq!(sql, format!("{}", out))
	}
```

Reply:
Ah thanks! Ones with LIMIT only! Added in 75ba014a 🙏